File: APIExamples/SensorData_API/java/ProcedureOperations.java
package javaapplication3;

import asua.sensordatasdk.client.ApiClient;
import asua.sensordatasdk.client.ApiException;
import asua.sensordatasdk.client.Configuration;
import asua.sensordatasdk.client.auth.*;
import asua.sensordatasdk.client.api.*;
import asua.sensordatasdk.client.model.*;
import java.util.List;

public class JavaApplication3 {

    public static void main(String[] args) {

        ApiClient defaultClient = Configuration.getDefaultApiClient();
        defaultClient.setBasePath("http://netigma.netcad.com.tr/sdrest");

        // Configure API key authorization: Authorization
        ApiKeyAuth authorization = (ApiKeyAuth) defaultClient.getAuthentication("Authorization");
        authorization.setApiKey("YOUR API KEY");

        ProcedureApi procedureApi = new ProcedureApi();

        try {
            //get all procedures
            List<ProcedureSummary> procedureSummarys = procedureApi.getAllProcedures();

            //get single procedure by it^s identifier
            Procedure procedure = procedureApi.getProcedure("identifier_example");

        } catch (ApiException e) {
            System.err.println("Exception when calling procedureApi");
            e.printStackTrace();
        }
    }
}

File: APIExamples/SensorData_API/dotnet/ProcedureOperations.cs
using ASUA.SensorDataApi.SDK.Api;
using ASUA.SensorDataApi.SDK.Client;
using ASUA.SensorDataApi.SDK.Model;
using System;
using System.Collections.Generic;

namespace ASUAWorkShop_APIExamples.SensorData_API
{
    class ProcedureOperationExamples
    {

        ApiClient apiClient = new ApiClient("http://netigma.netcad.com.tr/sdrest");
        ProcedureApi procedureApiInstance = null;

        public ProcedureOperationExamples()
        {
            Configuration conf = new Configuration(apiClient);
            conf.AddApiKey("Authorization", "YOUR_API_KEY");

            procedureApiInstance = new ProcedureApi(conf);
        }


        public void GetProcedures()
        {
            //Get single procedure by it'S identifier
            Procedure procedure = procedureApiInstance.GetProcedure("my_procedure_identifier");
            Console.WriteLine(procedure.ToString());

            //Get all procedures
            List<ProcedureSummary> procedureList = procedureApiInstance.GetAllProcedures();
            foreach (var item in procedureList)
            {
                Console.WriteLine(item.ToString());
            }
        }

    }
}
